File: typescript/ts_sdk/examples/wns.mjs
import {
  buildResolutionUrl,
  createHandleServiceEntry,
  parseUri,
  validateHandle,
  wns,
} from '../dist/index.js';

const [localPart, domain] = validateHandle('Alice.Example.COM');
console.log('Normalized handle:', `${localPart}.${domain}`);
console.log('Resolution URL:', buildResolutionUrl(localPart, domain));
console.log('WBA URI:', wns.buildUri(localPart, domain));
console.log('Parsed URI:', parseUri('wba://alice.example.com'));
console.log(
  'ANPHandleService entry:',
  createHandleServiceEntry('did:wba:example.com:user:alice', localPart, domain)
);

File: dart/example/wns.dart
import 'package:anp/anp.dart';

void main() {
  final parsed = validateHandle('alice.example.com');
  final uri = buildWbaUri(parsed.localPart, parsed.domain);
  final service = buildHandleServiceEntry(
    'did:wba:example.com:user:alice',
    parsed.localPart,
    parsed.domain,
  );
  print('$uri -> ${service.serviceEndpoint}');
}
